### DISCUSSION TITLE
Wie würde eine Minimum-Rechnung mit Profile.Comfort aussehen?

### DISCUSSION THREAD

**Discussion by DominikGeiler** [stephanstapel/ZUGFeRD-csharp#362] [General]
Hallo zusammen,
Mein Ziel ist es, eine valide ZUGFeRD-Rechnung zu erstellen, zunächst mit nur einer Position.
Anhand dieses Beispiels würde ich das Ganze dann ausbauen.

Wir haben hier ein eigenes ERP-System und müssen dieses nun aufgrund der Digitalisierung anpassen.
Die factur-x.xml habe ich bereits in die PDF eingebunden, ebenso die Schlüsselwörter in der PDF. Es fehlt "nur noch" die korrekte Zusammenstellung der XML.

**Comment by stephanstapel** [comment]
es gibt diverse Testfälle, schau da am besten rein!

**Reply by DominikGeiler** [reply]
Ja ich hab mich hier und in den offiziellen Dokumenten umgesehen.
Die Idee, die ich jetzt habe ist, dass ich die "factur-x.xml" für den EN16931 einfach abspeichere und als Vorlage lade und dann anpasse.
Oder ist diese Vorgehensweise generell nicht zu empfehlen?

In etwas so:
```


                // Lade Vorlage
                InvoiceDescriptor desc = InvoiceDescriptor.Load(pfad + System.IO.Path.DirectorySeparatorChar + "factur-x_en16931.xml");
                
                // Speichere die Factur-X XML
                string xmlPath = pfad + System.IO.Path.DirectorySeparatorChar + xmlfilename;

                desc.ActualDeliveryDate = DateTime.Now;
                desc.BusinessProcess = "Rechnung";
                desc.Buyer.City = "Offenburg";
                byte[] data = null;
                using (FileStream stream = new FileStream(xmlPath, FileMode.Create, FileAccess.Write))
                {
                    desc.Save(stream, ZUGFeRDVersion.Version22, Profile.Comfort);
                }
                
```
Doch wenn ich es so mache ist die XML nach dem speichern nicht mehr gültig

**Comment by landrix** [comment]
Deine Fragestellung deutet darauf hin, dass du noch komplett am Anfang der Thematik stehst. Ich würde Dir empfehlen einen Berater hinzuzuziehen, der Dich bei der Implementierung unterstützt oder eben die Testfälle zu studieren und dir das Wissen selbst anzueignen. Eine kostenlose Heranführung hier in den Diskussionen, damit wird niemand glücklich.

**Reply by stephanstapel** [reply]
@landrix : danke für die Anmerkung!
@DominikGeiler : lass' mich gerne wissen, wenn Du Beratungs-Unterstützung benötigst.

**Reply by DominikGeiler** [reply]
In der Tat bin ich ganz am Anfang und auch ein einzige Entwickler.
Ja das wäre wohl deutlich effizienter und schneller.
Was sind denn gute Anlaufstellen?
Ich bin nun dabei mir die Testfälle anzuschauen etc.

Hab nun gemerkt das wenn ich die vorlage nehme und nicht verädere ist die Datei valide.
Sobald ich aber mit InvoiceDescriptor öffne und dann speichere nicht mehr.

```
           // Lade Vorlage
              string xmlPath_vorlage = pfad + System.IO.Path.DirectorySeparatorChar + "factur-x_en16931.xml";
              InvoiceDescriptor desc = InvoiceDescriptor.Load(xmlPath_vorlage);
              
              // Speichere die Factur-X XML
              string xmlPath_facturx = pfad + System.IO.Path.DirectorySeparatorChar + "factur-x.xml";
              //desc.ActualDeliveryDate = DateTime.Now;
              //desc.BusinessProcess = "Rechnung";
              //desc.Buyer.City = "Offenburg";         

              File.Copy(xmlPath_vorlage, xmlPath_facturx, true);
              using (FileStream stream = new FileStream(xmlPath_facturx, FileMode.Create, FileAccess.Write))
              {
                  desc.Save(stream, ZUGFeRDVersion.Version22, Profile.Comfort);
              }

              PdfA3Processor.AddAttachmentToExistingPdf(targetPath, xmlPath_facturx, "");
```
              
@stephanstapel ja Unterstützung wäre nicht schlicht. Wie läuft das dann ab?

**Reply by stephanstapel** [reply]
@DominikGeiler : Schreibe mir gerne an: vorname punkt nachname at web punkt de
